apps/challenges/aws_utils.py

```diff
@@ -1061,6 +1061,257 @@ def scale_resources(challenge, worker_cpu_cores, worker_memory):
         return e.response
 
 
+def detach_policies_and_delete_role(challenge, iam):
```

Comment:
All these should be called inside each other in sequence (check the aws utils for EKS creation).
All of these should be celery tasks (check the aws utils for EKS creation).